lib/client-assets.php

```diff
@@ -289,6 +289,21 @@ function gutenberg_register_scripts_and_styles() {
 		)
 	);
 
+	// Add back compatibility for calls to wp.components.ServerSideRender.
+	wp_add_inline_script(
+		'wp-server-side-render',
+		implode(
+			"\n",
+			array(
+				'( function() {',
+				'	if ( wp && wp.components && wp.serverSideRender && ! wp.components.ServerSideRender ) {',
```

Comment:
An issue with this is if `wp-components` is enqueued after `wp-server-side-render`, it will never be shimmed. We could potentially resolve this by defining `wp-components` as a dependency of `wp-server-side-render` (optionally checking that it's at least registered).